src/components/inputs/text/NameTokenSelector/NameTokenSelector.tsx

```diff
@@ -1,27 +1,41 @@
-import { Tooltip } from 'antd';
-import { useState } from 'react';
+import { Pagination, PaginationProps, Tooltip } from 'antd';
+import { useEffect, useRef, useState } from 'react';
 
 import { useArweaveCompositeProvider } from '../../../../hooks';
-import { PDNTContract } from '../../../../services/arweave/PDNTContract';
+import { useGlobalState } from '../../../../state/contexts/GlobalState';
 import {
   ArweaveTransactionID,
   PDNTContractJSON,
   VALIDATION_INPUT_TYPES,
 } from '../../../../types';
+import { getAssociatedNames, isArweaveTransactionID } from '../../../../utils';
 import eventEmitter from '../../../../utils/events';
-import { CircleCheck, CirclePlus, CircleXIcon } from '../../../icons';
+import { CirclePlus, CloseIcon, HamburgerOutlineIcon } from '../../../icons';
 import { Loader } from '../../../layout';
 import ValidationInput from '../ValidationInput/ValidationInput';
+import './styles.css';
 
 function NameTokenSelector({
   selectedTokenCallback,
 }: {
   selectedTokenCallback: (id: ArweaveTransactionID | undefined) => void;
 }) {
   const arweaveDataProvider = useArweaveCompositeProvider();
+  const [{ pdnsSourceContract, walletAddress }] = useGlobalState();
 
   const [searchText, setSearchText] = useState<string>();
-  const [selectedNameToken, setSelectedNameToken] = useState<
+  const [tokens, setTokens] = useState<{
+    [x: string]: Pick<
+      PDNTContractJSON,
+      'name' | 'ticker' | 'owner' | 'controller'
+    > & { names?: string[] };
```

Comment:
create a type as this is used in a few spots - can be created in this component as it's not needed anywhere else

Reply:
updated in https://github.com/ar-io/pdns-react/pull/229/commits/0b63d7bccbf324d95029f2a91e4a74e4e9116c0e